packages/node/src/behaviors/thermostat/ThermostatServer.ts

```diff
@@ -4,14 +4,1549 @@
+    Thermostat.Feature.Cooling,
+    Thermostat.Feature.Occupancy,
+    Thermostat.Feature.AutoMode,
+    Thermostat.Feature.Presets,
+    Thermostat.Feature.Setback,
+);
+
+// Enhance Schema to define conformance for some of the additional state attributes
+const schema = ThermostatBehaviorLogicBase.schema!.extend({
+    children: [
+        FieldElement({
+            name: "PersistedPresets",
+            type: "list",
+            conformance: "[PRES]",
+            quality: "N",
+            children: [FieldElement({ name: "entry", type: "PresetStruct" })],
+        }),
+    ],
+});
+
 /**
  * This is the default server implementation of {@link ThermostatBehavior}.
  *
  * The Matter specification requires the Thermostat cluster to support features we do not enable by default. You should
  * use {@link ThermostatServer.with} to specialize the class for the features your implementation supports.
+ * We implement all features beside the following:
+ * * MatterScheduleConfiguration: This feature is provisional.
+ * * ScheduleConfiguration: This feature is deprecated and not allowed to be enabled.
+ * * The use of the "setpointHoldExpiryTimestamp" attribute is currently not supported.
+ *
+ * This implementation mainly provides all validation and base logic required by the Matter specification.
+ * It implements some thermostat logic, partially beyond Matter specification definition, notably:
+ * * Adjust the setpoints when a preset is activated
+ * If this behavior is not desired, you can override the setActivePresetRequest method but should call
+ * handleSetActivePresetRequest() to ensure compliance with the specification.
+ *
+ * The implementation also adds enhanced system mode logic that can be enabled by setting the state field
+ * useAutomaticModeManagement to true. When enabled, the thermostat will:
+ * * Adjust the thermostat running mode when in Auto system mode and the Setback feature is also supported
+ * * Determine the system mode and/or running mode based on temperature changes
+ *
+ * For local temperature or occupancy values we check if there is a local cluster available on the same endpoint and use
+ * them, alternatively raw measurements can be set in the states externalMeasuredIndoorTemperature and
+ * externallyMeasuredOccupancy. The OutdoorTemperature can be set directly on the attribute if supported.
+ * The RemoteSensing attribute need to be set correctly as needed by the developer to identify the measurement source.
+ *
+ * The following custom events are provided:
+ * * calibratedTemperature$Changed: Emitted when the measured local temperature changes including any calibration applied. This event is mainly useful when the localTemperatureNotExposed feature is used.
+ *
+ * Important note: To access the current local temperature (including all calibrations applied) please use
+ * this.internal.localTemperature because the localTemperature attribute in state might be null depending on the
+ * configured features.
```

Comment:
Maybe TODO or NOTE here that documents that we implement general-purpose "atomic" functionality inline in this cluster because it's the only cluster that uses it and we would like to wait and see how the feature evolves before making it general purpose, and/or link to `AtomicWriteHandler.ts` for more information.